app/services/doc_auth_router.rb

```diff
@@ -185,20 +189,36 @@ def self.client(warn_notifier: nil)
           sharpness_threshold: IdentityConfig.store.doc_auth_error_sharpness_threshold,
           glare_threshold: IdentityConfig.store.doc_auth_error_glare_threshold,
         ),
+        vendor_discriminator: vendor_discriminator,
       )
     when 'mock'
       DocAuthErrorTranslatorProxy.new(
         DocAuth::Mock::DocAuthMockClient.new(
           warn_notifier: warn_notifier,
         ),
+        vendor_discriminator: vendor_discriminator,
       )
     else
-      raise "#{doc_auth_vendor} is not a valid doc auth vendor"
+      raise "#{doc_auth_vendor(discriminator: vendor_discriminator)} is not a valid doc auth vendor"
     end
   end
   # rubocop:enable Layout/LineLength
 
-  def self.doc_auth_vendor
+  def self.doc_auth_vendor(discriminator: nil)
+    if IdentityConfig.store.doc_auth_vendor_randomize
```

Comment:
Minor: Does this need to exist at all as a standalone config value, or could it be inferred by `doc_auth_vendor_randomize_alternate_vendor.present?` or `doc_auth_vendor_randomize_percent > 0`?

Reply:
Sorry, this one got forgotten. It could but I like having the explicit flag for it being on or off as a feature and the configuration of the feature are separate from it.

Reply:
> Sorry, this one got forgotten. It could but I like having the explicit flag for it being on or off as a feature and the configuration of the feature are separate from it.

Sounds fine to me 👍